packages/material-ui/src/Slider/Slider.js

```diff
@@ -1,1035 +1,312 @@
 import * as React from 'react';
-import PropTypes from 'prop-types';
-import clsx from 'clsx';
-import { chainPropTypes } from '@material-ui/utils';
-import withStyles from '../styles/withStyles';
+import { css, cx } from 'emotion';
```

Comment:
Is there a way we could avoid using the `css` helper (for interop with styled-components)? Have it handled by the styled() API

Reply:
The `css` is used for resolving the definitions inside the theme (styleOverrides and variant style). We can move the css directly in the theme definition, that way users of different libraries would be able to use different `css` utilities. 

```js
const theme = {
  components: { 
    MuiSlider: {
      styleOverrides: { 
        root: css({...}), // <-- use the css directly here..
      },
      variants: [
        {
           props: { orientation: 'vertical', color: 'primary'},
           style: css({...}), 
    }
}
```

Should we do this change?

Note:
There is another challenge here - how do we make sure that they will be applied in the correct order (for emotion and cx is predictable, for styled-components I am not sure, will have to test this).

Reply:
> The css is used for resolving the definitions inside the theme

Does it mean that developers don't have access to the props of the component in the theme (as they do now)? 

> how do we make sure that they will be applied in the correct order

We can provide an array of styles to `styled()`, I assume the order of the array determines the correct order.

Reply:
Solved, we are using `styled` exclusively now